argon2/src/params.rs

```diff
@@ -384,8 +387,27 @@ pub struct ParamsBuilder {
 
 impl ParamsBuilder {
     /// Create a new builder with the default parameters.
-    pub fn new() -> Self {
-        Self::default()
+    pub const fn new() -> Self {
+        Self::DEFAULT
+    }
+    /// Create a new builder with the provided parameters.
+    /// This function exists to allow for const construction of ParamsBuilder with custom parameters.
+    pub const fn new_params(
+        m_cost: u32,
+        t_cost: u32,
+        p_cost: u32,
+        keyid: Option<KeyId>,
+        data: Option<AssociatedData>,
+        output_len: Option<usize>,
+    ) -> Self {
+        Self {
+            m_cost,
+            t_cost,
+            p_cost,
+            keyid,
+            data,
+            output_len,
+        }
     }
```

Comment:
This seems like it defeats the point of having a builder. The reason it exists is so you don't have to pass a big bag of parameters like this.

Reply:
But then why have a builder?